fabric-events-interaction-v0/src/main/java/net/fabricmc/fabric/mixin/event/interaction/MixinServerPlayerInteractionManager.java

```diff
@@ -81,4 +86,30 @@ public void interactItem(ServerPlayerEntity player, World world, ItemStack stack
 			return;
 		}
 	}
+
+	@Inject(at = @At(value = "INVOKE", target = "Lnet/minecraft/block/Block;onBreak(Lnet/minecraft/world/World;Lnet/minecraft/util/math/BlockPos;Lnet/minecraft/block/BlockState;Lnet/minecraft/entity/player/PlayerEntity;)V"), method = "tryBreakBlock", locals = LocalCapture.CAPTURE_FAILHARD, cancellable = true)
+	private void breakBlock(BlockPos pos, CallbackInfoReturnable<Boolean> cir, BlockState state, BlockEntity entity, Block block) {
+		boolean result = PlayerBlockBreakEvents.BEFORE.invoker().beforeBlockBreak(this.world, this.player, pos, state, entity);
+
+		if (!result) {
+			PlayerBlockBreakEvents.CANCELED.invoker().onBlockBreakCancel(this.world, this.player, pos, state, entity);
```

Comment:
Since the cancel event is fired first before the normal vanilla packet, wouldn't this cause a double update on some blocks?

Maybe only call vanilla logic if the result of cancel even explicitly says to fallback to vanilla logic (likely true return value).

Reply:
The double update is fine. This is for modded blocks which would need the update to extend further than a three by three, since the client thinks the block was broken.